juju/juju
# feat: remove charm url view

To accommodate annotations in charms, we introduced a charm_url view, to allow backward compatibility. We must stop using that as it can confuse people when trying to wire up a charm in various parts of the code.

This solution although not pretty is more correct.

The non-pretty part is that we're using ParseURL in the core package. Once we move that package to the domain, this becomes a lot more palatable.


## QA steps

Ensure that we can bootstrap, there isn't any test for charm annotations. We know it's broken, because of `names.CharmTag` doesn't support charmhub charms.

```sh
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
```


## Links


Follow up bug: https://bugs.launchpad.net/juju/+bug/2089149
